pkg/planner/core/plan_cache_test.go

```diff
@@ -852,16 +852,22 @@ func testRandomPlanCacheCases(t *testing.T,
 	tk := testkit.NewTestKit(t, store)
 	prepFunc(tk)
 
-	// prepared plan cache
+	// nonprepared plan cache
 	for _, q := range queryFunc(true) {
 		tk.MustExec("set tidb_enable_non_prepared_plan_cache=0")
 		result1 := tk.MustQuery(q).Sort()
 		tk.MustExec("set tidb_enable_non_prepared_plan_cache=1")
+
+		// the first execution caches the plan
 		result2 := tk.MustQuery(q).Sort()
 		require.True(t, result1.Equal(result2.Rows()))
+
+		// the second execution uses the cache
+		result2 = tk.MustQuery(q).Sort()
```

Comment:
And that it was from the cache
```suggestion
		result2 = tk.MustQuery(q).Sort()
		tk.MustQuery(`select @@last_plan_from_cache`).Check(testkit.Rows("1"))
```

Reply:
Some of the queries are not cacheable.  I will remove this comment.